packages/plugin-core/src/commands/ConvertLink.ts

```diff
@@ -16,31 +44,310 @@ export class ConvertLinkCommand extends BasicCommand<
+          text = reference.ref;
+        }
+        break;
+      }
+      case "Note name": {
+        text = _.last(parsedLink.value!.split("."));
+        if (
+          reference.refType === "hashtag" ||
+          reference.refType === "usertag"
+        ) {
+          text = _.last(reference.ref.split("."));
+        }
+        break;
+      }
+      case "Prompt": {
+        text = this.promptBrokenLinkUserInput();
+        break;
+      }
+      case "Change destination": {
+        const resp = await this.lookupNewDestination();
+        if (_.isUndefined(resp)) {
+          break;
+        }
+        text = NoteUtils.createWikiLink({
+          note: resp?.selectedItems[0] as NoteProps,
+          alias: { mode: "title" },
+        });
+        break;
+      }
+      default: {
+        assertUnreachable();
+      }
+    }
+    return text;
+  }
+
+  async promptConfirmation(opts: { title: string; noConfirm?: boolean }) {
+    const { title, noConfirm } = opts;
+    if (noConfirm) return true;
+    const options = ["Proceed", "Cancel"];
+    const resp = await VSCodeUtils.showQuickPick(options, {
+      placeHolder: "Proceed",
+      ignoreFocusOut: true,
+      title,
+    });
+    return resp === "Proceed";
+  }
+
+  async prepareValidLinkOperation(reference: getReferenceAtPositionResp) {
+    const { refType, range, ref } = reference;
+    switch (refType) {
+      case "hashtag":
+      case "usertag": {
+        const shouldProceed = await this.promptConfirmation({
+          title: `Convert ${refType} to wikilink?`,
+        });
+        if (shouldProceed) {
+          return {
+            range,
+            text: `[[${ref}]]`,
+          };
+        }
+        break;
+      }
+      case "wiki": {
+        let tagType;
+        if (ref.startsWith("user")) {
+          tagType = "usertag";
+        } else if (ref.startsWith("tags")) {
+          tagType = "hashtag";
+        }
+
+        if (_.isUndefined(tagType)) {
+          throw this.noAvailableOperationError;
+        }
+
+        const shouldProceed = await this.promptConfirmation({
+          title: `Convert wikilink to ${tagType}?`,
+        });
+        if (shouldProceed) {
+          const label = _.drop(ref.split(".")).join(".");
+          const text = tagType === "usertag" ? `@${label}` : `#${label}`;
+          return {
+            range,
+            text,
+          };
+        }
+        break;
+      }
+      case "fmtag":
+      case "refv2": {
+        throw this.noAvailableOperationError;
+      }
+      default: {
+        assertUnreachable();
+      }
+    }
+    throw new DendronError({
+      message: "cancelled.",
+    });
+  }
+
+  async gatherInputs(): Promise<CommandOpts> {
+    const { engine } = getDWorkspace();
+    const { vaults, wsRoot, notes } = engine;
+    const editor = VSCodeUtils.getActiveTextEditor() as TextEditor;
+    const { document, selection } = editor;
+    const reference = getReferenceAtPosition(document, selection.start);
```

Comment:
heads up that this takes a `allowInCodeBlocks` option. might make sense for this command to support